src/main/java/io/cdap/plugin/gcp/bigtable/source/BigtableSource.java

```diff
@@ -0,0 +1,249 @@
+ */
+@Plugin(type = "batchsource")
+@Name(BigtableSource.NAME)
+@Description("This source reads data from Google Cloud Bigtable. " +
+  "Cloud Bigtable is Google's NoSQL Big Data database service.")
+public final class BigtableSource extends BatchSource<ImmutableBytesWritable, Result, StructuredRecord> {
+  public static final String NAME = "Bigtable";
+  private static final Logger LOG = LoggerFactory.getLogger(BigtableSource.class);
+  private final BigtableSourceConfig config;
+  private Schema outputSchema;
+
+  public BigtableSource(BigtableSourceConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer configurer) {
+    super.configurePipeline(configurer);
+    config.validate();
+    Schema configuredSchema = config.getSchema();
+    if (null != configuredSchema && config.connectionParamsConfigured()) {
+      Configuration conf = getConfiguration();
+      validateOutputSchema(conf);
+    }
+
+    configurer.getStageConfigurer().setOutputSchema(configuredSchema);
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) {
+    config.validate();
+    Configuration conf = getConfiguration();
+    validateOutputSchema(conf);
+
+    // Both emitLineage and setOutputFormat internally try to create an external dataset if it does not already exists.
+    // We call emitLineage before since it creates the dataset with schema.
+    emitLineage(context);
+    context.setInput(Input.of(config.referenceName, new SourceInputFormatProvider(TableInputFormat.class, conf)));
+  }
+
+  private Configuration getConfiguration() {
+    try {
+      Configuration conf = HBaseConfiguration.create();
+      BigtableConfiguration.configure(conf, config.getProject(), config.instance);
+      conf.set(BigtableOptionsFactory.BIGTABLE_RPC_TIMEOUT_MS_KEY, "60000");
+      conf.setBoolean(TableInputFormat.SHUFFLE_MAPS, true);
+      conf.set(TableInputFormat.INPUT_TABLE, config.table);
+      Scan scan = getConfiguredScanForJob();
+      conf.set(TableInputFormat.SCAN, TableMapReduceUtil.convertScanToString(scan));
+      return conf;
+    } catch (IOException e) {
+      throw new InvalidStageException("Failed to prepare configuration for job", e);
+    }
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    super.initialize(context);
+    outputSchema = context.getOutputSchema();
+  }
+
+  /**
+   * Converts <code>JsonObject</code> to <code>StructuredRecord</code> for every record
+   * retrieved from the Bigtable.
+   *
+   * @param input   input record
+   * @param emitter emitting the transformed record into downstream nodes.
+   */
+  @Override
+  public void transform(KeyValue<ImmutableBytesWritable, Result> input, Emitter<StructuredRecord> emitter) {
+    try {
+      StructuredRecord record = convertInputToRecord(input);
+      emitter.emit(record);
+    } catch (Exception e) {
+      switch (config.getErrorHandling()) {
```

Comment:
I didn't see this property in the reference doc.

Reply:
This is a name of the enum class. Property is called onError. This getter just parses user-defined property and returns enum.

Reply:
Can you make sure it's included in the reference doc with a description?